pkg/session_recording/player.jsx

```diff
@@ -0,0 +1,823 @@
+                                    /* Window (8, 9) */
+                                    "=(\\d+)x(\\d+)|" +
+                                    /* End of string */
+                                    "$",
+                                    /* Continue after the last match only */
+                                    /* FIXME Support likely sparse */
+                                    "y"
+            );
+            /* List of matches to apply when loading the buffer from Journal */
+            this.matchList = matchList;
+            /*
+             * An array of two-element arrays (tuples) each containing a
+             * packet index and a deferred object. The list is kept sorted to
+             * have tuples with lower packet indices first. Once the buffer
+             * receives a packet at the specified index, the matching tuple is
+             * removed from the list, and its deferred object is resolved.
+             * This is used to keep users informed about packets arriving.
+             */
+            this.idxDfdList = [];
+            /* Last seen message ID */
+            this.id = 0;
+            /* Last seen time position */
+            this.pos = 0;
+            /* Last seen window width */
+            this.width = null;
+            /* Last seen window height */
+            this.height = null;
+            /* List of packets read */
+            this.pktList = [];
+            /* Error which stopped the loading */
+            this.error = null;
+            /* The journalctl reading the recording */
+            this.journalctl = Journal.journalctl(
+                                this.matchList,
+                                {count: "all", follow: false});
+            this.journalctl.fail(this.handleError);
+            this.journalctl.stream(this.handleStream);
+            this.journalctl.done(this.handleDone);
+            /*
+             * Last seen cursor of the first, non-follow, journalctl run.
+             * Null if no entry was received yet, or the second run has
+             * skipped the entry received last by the first run.
+             */
+            this.cursor = null;
+            /* True if the first, non-follow, journalctl run has completed */
+            this.done = false;
+        }
+
+        /*
+         * Return a promise which is resolved when a packet at a particular
+         * index is received by the buffer. The promise is rejected with a
+         * non-null argument if an error occurs or has occurred previously.
+         * The promise is rejected with null, when the buffer is stopped. If
+         * the packet index is not specified, assume it's the next packet.
+         */
+        awaitPacket(idx) {
+            let i;
+            let idxDfd;
+
+            /* If an error has occurred previously */
+            if (this.error !== null) {
+                /* Reject immediately */
+                return $.Deferred().reject(this.error).promise();
```

Comment:
One thing I've found after conversations with Cockpit's team - they are planning to get rid of jQuery, I think we can keep, but maybe we might look forward using underscore.js or something else.
Anyway, our code now is related on jQuery a lot.

Reply:
I see. Can we use the [standard Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) instead?

Reply:
Yes, since we are using babel and ES6.

Reply:
Alright, I'll see if I can switch to Promises quickly.

Reply:
I read about Promises and found out that they don't naturally support the approach I use and switching to them would require some hops and a bit of ugly code. So, I think it is better to leave that for later. Perhaps we'll have a better idea by the time jQuery is going away for real.

Is that OK with you? If yes, I'll merge this as it is.

Reply:
Yes, I think it's okay, anyway we are using jQuery a lot.